gss-serv-krb5.c

```diff
@@ -100,11 +127,38 @@ ssh_gssapi_krb5_userok(ssh_gssapi_client *client, char *name)
 	}
 	if (krb5_kuserok(krb_context, princ, name)) {
```

Comment:
IMO this function would be clearer if it shortcircuits when preconditions fail, i.e.

```C
	if (!krb5_kuserok(krb_context, princ, name))
		goto out;
	if (options.gss_indicators == NULL)  {
		retval = 1;
		goto out;
	}
	if (!client->indicators) {
		logit("GSSAPI authentication indicators enforced "
			"but not matched. krb5 principal %s denied",
		(char *)client->displayname.value);
		goto out;
	}
...
 out:
	if (retval == 1)  {
		logit("Authorized to %s, krb5 principal %s (%s)",
		    name, (char *)client->displayname.value, errmsg);
	}
	krb5_free_principal(krb_context, princ);
	return retval;
```

Reply:
Thanks, it makes sense. I rewrote the logic to reduce if-conditions.